flang/test/Driver/print-supported-extensions-arm.f90

```diff
@@ -0,0 +1,34 @@
+! REQUIRES: arm-registered-target
+
+! RUN: %flang --target=arm-linux-gnu --print-supported-extensions 2>&1 \
+! RUN:   | FileCheck --strict-whitespace --implicit-check-not=FEAT_ %s
+
+! CHECK: All available -march extensions for ARM
```

Comment:
> IMO, it would be sufficient to only test this line and refer folks to Clang for the rest. Just to avoid duplicating tests. WDYT?

I would be ok with this if the output was produced entirely from code that lived in `clang/` and we just added hooks to drive it. In this case, while the actual output comes from `llvm/`, it is driven by the `printSupportedExtensions()` function that was added in this PR to `flang/tools/flang-driver/fc1_main.cpp`. 

I think we could the driving code, like `printSupportedExtensions` to `libclangDriver` (which `flang-driver` already uses). That way, we only need to keep a single copy of the tests since it is only the one code that drives it. I could do that for this PR, and then an "effectively NFC" one do the same thing for `printSupportedCPUs()`. 

Thoughts?

Reply:
If the implementation in Clang and Flang are identical, then it should be re-used rather than re-implemented. Such "hooks" usually end up in:
*  https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/ToolChains/CommonArgs.cpp.

We are definitely missing some better way to organise these things.

Anyway, thanks for working on this!

Reply:
It makes sense to shared these and I'll change this PR to do that. Sharing the `printSupportedCPUs` (which is currently duplicated) will be done separately. 

Is `clang/lib/Driver/ToolChains/CommonArgs.cpp` an appropriate place for the shared code? Is it not intended to be "private" to the toolchains? I don't see anything more appropriate, but we're probably in uncharted territory here.

Reply:
I don't have a better suggestion. I do recall similar situation in the past and being recommended to use that file. I do feel though that it would be worthwhile creating a dedicated cpp file for this kind of things. 

Once you update this PR, we can ping Clang driver code-owner.